numba_cuda/numba/cuda/core/interpreter.py

```diff
@@ -0,0 +1,3588 @@
+            self.store(ir.Expr.cast(self.get(retval), loc=self.loc), castval)
+            ret = ir.Return(self.get(castval), loc=self.loc)
+            self.current_block.append(ret)
+    elif PYVERSION in ((3, 10), (3, 11)):
+        pass
+    else:
+        raise NotImplementedError(PYVERSION)
+
+    if PYVERSION in ((3, 13),):
+
+        def op_TO_BOOL(self, inst, val, res):
+            self.store(self.get(val), res)  # TODO: just a lazy hack
+
+    elif PYVERSION in ((3, 10), (3, 11), (3, 12)):
+        pass
+    else:
+        raise NotImplementedError(PYVERSION)
+
+    def op_COMPARE_OP(self, inst, lhs, rhs, res):
+        if PYVERSION in ((3, 13),):
+            op = dis.cmp_op[inst.arg >> 5]
+            # TODO: fifth lowest bit now indicates a forced version to bool.
+        elif PYVERSION in ((3, 12),):
+            op = dis.cmp_op[inst.arg >> 4]
+        elif PYVERSION in ((3, 10), (3, 11)):
+            op = dis.cmp_op[inst.arg]
+        else:
+            raise NotImplementedError(PYVERSION)
+        if op == "in" or op == "not in":
+            lhs, rhs = rhs, lhs
+
+        if op == "not in":
+            self._binop("in", lhs, rhs, res)
+            tmp = self.get(res)
+            out = ir.Expr.unary("not", value=tmp, loc=self.loc)
+            self.store(out, res)
+        elif op == "exception match":
+            gv_fn = ir.Global(
+                "exception_match",
+                eh.exception_match,
+                loc=self.loc,
+            )
+            exc_match_name = "$exc_match"
+            self.store(value=gv_fn, name=exc_match_name, redefine=True)
+            lhs = self.get(lhs)
+            rhs = self.get(rhs)
+            exc = ir.Expr.call(
+                self.get(exc_match_name),
+                args=(lhs, rhs),
+                kws=(),
+                loc=self.loc,
+            )
+            self.store(exc, res)
+        else:
+            self._binop(op, lhs, rhs, res)
+
+    def op_IS_OP(self, inst, lhs, rhs, res):
+        # invert if op case is 1
+        op = "is not" if inst.arg == 1 else "is"
+        self._binop(op, lhs, rhs, res)
+
+    def op_CONTAINS_OP(self, inst, lhs, rhs, res):
+        lhs, rhs = rhs, lhs
+        self._binop("in", lhs, rhs, res)
+        # invert if op case is 1
+        if inst.arg == 1:
+            tmp = self.get(res)
+            out = ir.Expr.unary("not", value=tmp, loc=self.loc)
+            self.store(out, res)
+
+    def op_BREAK_LOOP(self, inst, end=None):
+        if end is None:
+            loop = self.syntax_blocks[-1]
+            assert isinstance(loop, ir.Loop)
+            end = loop.exit
+        jmp = ir.Jump(target=end, loc=self.loc)
+        self.current_block.append(jmp)
+
+    def _op_JUMP_IF(self, inst, pred, iftrue):
+        brs = {
+            True: inst.get_jump_target(),
+            False: inst.next,
+        }
+        truebr = brs[iftrue]
+        falsebr = brs[not iftrue]
+
+        name = "bool%s" % (inst.offset)
```

Comment:
Are you sure you've vendored the interpreter from a recent commit? In the commit you linked to, it seems that this line is 

```suggestion
        name = "$bool%s" % (inst.offset)
```

which is a fix for a debuginfo issue that we upstreamed. (observed from https://github.com/numba/numba/blob/1358274054f88c19838ba11543b8f87b05cf0632/numba/core/interpreter.py#L3059)

Maybe I'm getting something backwards / wrong here - if so, please point me in the correct direction! :slightly_smiling_face:

Reply:
Hm, I checked the commit hash and copied it over directly🤔 I will pull the latest version and copy it in again. How does that sound?

Reply:
That was about the only difference between my last commit and the most recent interpreter.py:
```diff
diff --git a/numba_cuda/numba/cuda/core/interpreter.py b/numba_cuda/numba/cuda/core/interpreter.py
index 6c6b5c5..abf8216 100644
--- a/numba_cuda/numba/cuda/core/interpreter.py
+++ b/numba_cuda/numba/cuda/core/interpreter.py
@@ -16,11 +16,13 @@ from numba.core.utils import (
     PYVERSION,
     BINOPS_TO_OPERATORS,
     INPLACE_BINOPS_TO_OPERATORS,
+    _lazy_pformat,
 )
 from numba.core.byteflow import Flow, AdaptDFA, AdaptCFA, BlockKind
 from numba.core.unsafe import eh
 from numba.cpython.unsafe.tuple import unpack_single_tuple
 
+
 if PYVERSION in ((3, 12), (3, 13)):
     # Operands for CALL_INTRINSIC_1
     from numba.core.byteflow import CALL_INTRINSIC_1_Operand as ci1op
@@ -1420,7 +1422,9 @@ class Interpreter(object):
             self.arg_count,
             self.arg_names,
         )
-        _logger.debug(func_ir.dump_to_string())
+        _logger.debug(
+            _lazy_pformat(func_ir, lazy_func=lambda x: x.dump_to_string())
+        )
 
         # post process the IR to rewrite opcodes/byte sequences that are too
         # involved to risk handling as part of direct interpretation
@@ -3183,7 +3187,7 @@ class Interpreter(object):
         truebr = brs[iftrue]
         falsebr = brs[not iftrue]
 
-        name = "bool%s" % (inst.offset)
+        name = "$bool%s" % (inst.offset)
         gv_fn = ir.Global("bool", bool, loc=self.loc)
         self.store(value=gv_fn, name=name)
```

Reply:
These are the same differences that I saw - thanks for confirming!